winsup/cygwin/exceptions.cc

```diff
@@ -1547,10 +1548,7 @@ sigpacket::process ()
   if (have_execed)
     {
       sigproc_printf ("terminating captive process");
-      if ((sigExeced = si.si_signo) == SIGINT)
-        kill_process_tree (GetProcessId (ch_spawn), sigExeced = si.si_signo);
-      else
-        TerminateProcess (ch_spawn, sigExeced = si.si_signo);
+      exit_process (ch_spawn, 128 + (sigExeced = si.si_signo));
```

Comment:
Might be wise to fall back to `kill_process_tree` if `exit_process` fails here. Otherwise, we've stolen the power of Ctrl+C as a reliable process termination utility.